### PR TITLE
feat: add --check option to nargo fmt for dry-run formatting verification

### LINKED ISSUE #3520: Add `nargo fmt --check` functionality
### Problem

We would like to integrate `nargo fmt`  in CI in aztec-packages but to do that we would need to run `nargo fmt` in "check-only mode" (not modifying the files and just throwing instead).

### Happy Case

`nargo fmt --check` exits with exit code 1 when formatting issues are spotted.

### Alternatives Considered

_No response_

### Additional Context

_No response_

### Would you like to submit a PR for this Issue?

No

### Support Needs

_No response_</div>

### PR DESCRIPTION
**Pull request by f01dab1e**
# Description

```bash
❯ cargo r -- fmt --check
Differences (-original|+formatted):
 fn main() {
     let _callStackItem3 = context.call_private_function(
-        
         inputs.call_context.storage_contract_address,
         get_note_zero_fn_selector,
         [owner]
     );


❯ echo $?
1
```

```bash
❯ cargo r -- fmt --check
No formatting changes were detected

❯ echo $?
0
```

## Problem

Resolves #3520 

## Documentation

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: tooling/nargo_cli/src/cli/fmt_cmd.rs [hunk 1]
```diff
@@ -13,9 +13,15 @@ use super::NargoConfig;
 
 /// Format the Noir files in a workspace
 #[derive(Debug, Clone, Args)]
-pub(crate) struct FormatCommand {}
+pub(crate) struct FormatCommand {
+    /// Run nargofmt in check mode
```
Comment (f01dab1e) [inline tooling/nargo_cli/src/cli/fmt_cmd.rs:17]:
```suggestion
    /// Run noirfmt in check mode
```

File: tooling/nargo_cli/src/cli/fmt_cmd.rs [hunk 2]
```diff
@@ -53,16 +61,31 @@ pub(crate) fn run(_args: FormatCommand, config: NargoConfig) -> Result<(), CliEr
...
-            );
+            let original = file_manager.fetch_file(file_id).source();
+            let formatted = nargo_fmt::format(original, parsed_module, &config);
 
-            std::fs::write(entry.path(), source)
+            if check_mode {
+                exit_code_one = original != formatted;
+
+                let diff = similar_asserts::SimpleDiff::from_str(
+                    &original,
```
Comment (f01dab1e) [inline tooling/nargo_cli/src/cli/fmt_cmd.rs:71]:
```suggestion
                    original,
```

**Comment by kevaundray** [comment]
> Invisible differences: the two values are the same in string form.

If this is the output when there are no formatting changes, we should change it to something like: "No formatting changes were detected" which is eaiser to read for me.

Prettier says something like "All matched files use Prettier code style!" for example

File: tooling/nargo_cli/src/cli/fmt_cmd.rs [hunk 3]
```diff
@@ -53,16 +61,40 @@ pub(crate) fn run(_args: FormatCommand, config: NargoConfig) -> Result<(), CliEr
...
+            } else {
+                std::fs::write(entry.path(), formatted)
+            }
         })
         .map_err(|error| CliError::Generic(error.to_string()))?;
     }
+
+    if check_exit_code_one {
+        std::process::exit(1);
+    } else {
```
Comment (f01dab1e) [inline tooling/nargo_cli/src/cli/fmt_cmd.rs:94]:
```suggestion
    } else if check_mode {
```

**Review by kevaundray — APPROVED** [review]

**Pull request merged**